src/gf.c

```diff
@@ -667,6 +669,15 @@ JL_DLLEXPORT void jl_update_codeinst(
     if ((const_flags & 1) != 0) {
         assert(codeinst->rettype_const);
         jl_atomic_store_release(&codeinst->invoke, jl_fptr_const_return);
+        // Emit a `precompile` since this code's compilation is now complete. It was just inferred and it will require
+        // no codegen since it returns a constant value.
+        jl_method_instance_t *mi = jl_get_ci_mi(codeinst);
+        if (jl_is_method(mi->def.value) && jl_isa_compileable_sig((jl_tupletype_t *)mi->specTypes, mi->sparam_vals, mi->def.method)) {
+            // This isa_compileable_sig check is required because `precompile(...)` will reject the signature otherwise,
+            // assuming it is too wide to compile to a good result. That happens even though we know a posteriori that
+            // the signature compiles to a function with excellent performance (a const-return function).
+            record_precompile_statement(mi, 0.0, 0);
```

Comment:
This probably works but it makes me nervous, since all the other calls to this are from `jl_compile_method_internal`. Is there any way to arrange for this call to also come from that function?